### DISCUSSION TITLE
Export Categories Unread Links?

### DISCUSSION THREAD

**Discussion by Fogest** [FreshRSS/FreshRSS#3437] [Q&A]
Hello,

I have been enjoying FreshRSS for awhile and recently I started following some music based YouTube channel RSS feeds. I am trying to make the process of downloading these videos with youtube-dl a bit easier but am currently finding it a bit tedious.

Is there anyway I could from the UI or command line export a list of unread links from a category? I currently am just going through and manually copying and pasting links into a .txt file and running youtube-dl on that which works, but the tedious part is manually clicking through each FreshRSS entry and copying and pasting the link into the text file. 

Any suggestions for would be appreciated <3

Edit:

Just for a quick hacky way I realized I could just throw together a userscript for this. I already have Tampermonkey so it wouldn't be a big deal to just have a userscript for this.

For example this works when I'm in the category of unread youtube videos to grab a list of links:
```javascript
let l = document.links;
for(let i = 0; i < l.length; i++) {
    if(l[i].href.includes("youtube")) console.log(l[i].href);
}
```
Of course I'd clean up that code for making it into a userscript. Wondering if this is the best/easiest method to go about it?

**Comment by Fogest** [comment]
I've made a userscript that can do it. The code can be found here: https://github.com/Fogest/FreshRSS-YouTube-to-Clipboard/blob/master/freshrss-youtube-clipboard.user.js

Greasemonkey/Tampermonkey Direct Install Link: https://github.com/Fogest/FreshRSS-YouTube-to-Clipboard/raw/master/freshrss-youtube-clipboard.user.js

**Comment by aledeg** [comment]
I can make a shameless plug by promoting my firefox extension. It helps me solve the kind of problem you're facing.
https://github.com/aledeg/linkdump

**Reply by Fogest** [reply]
Looks pretty cool! Unfortunately I use Google Chrome though. The userscript I made is definitely nowhere near as comprehensive as your add-on. It does make me happy to see I am not the only one who had this same use case!

**Reply by aledeg** [reply]
Thank you! I am sorry to hear you're using Chrome since I am not planning to support it. I've checked into it but there is a lot of limitation compared to Firefox.

**Reply by Fogest** [reply]
I'm curious what an example of these limitations may be? I know Chrome imposes some more strict security in some circumstances but the things you are currently doing seem to be things Chrome supports. The stuff you have seems to be stuff that could also be done at the userscript level which makes me think Chrome shouldn't have an issue with it. 

But I am not sure how things are currently setup, depending on how it is coded it may be a lot of work to support Chrome and I can understand why you'd want to avoid Chrome then :)

Luckily for me I don't need to do anything more complex with the link saving, but if I did I'd likely use Firefox when needed and use your add-on so I'll definitely keep this saved for the future!

**Reply by aledeg** [reply]
One of the limitation is I can add only one type of action. I cannot have both an action in the address bar and an action in the tool bar. Which is weird :/
Regarding of the code itself, I think what I use is pretty standard so I guess, it could work smoothly. But I can't remove the tool bar action since it's where I can get the scraped links. And I don't want to remove the address bar action since it allows me to have a simple workflow.
